### PR TITLE
ld preloadable libdatadog trace sidecar

### PR DESCRIPTION
**Pull request by pawelchcki**
# What does this PR do?

A brief description of the change being made with this pull request.

# Motivation

What inspired you to submit this pull request?

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

Describe here in detail how the change can be validated.

### CODE REVIEW

File: sidecar/src/connections.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ) -> Poll<std::io::Result<()>> {
+        self.project().inner.poll_read(cx, buf)
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct Tracker {
+    count: Arc<AtomicUsize>,
+    notifier: Arc<tokio::sync::Notify>,
+}
```
Comment (paullegranddc) [inline sidecar/src/connections.rs:105]: One arc is better than two maybe?

```suggestion
struct TrackerInner {
    count: AtomicUsize,
    notifier: tokio::sync::Notify,
}

pub struct Tracker {
  inner: Arc<TrackerWatcherInner>
}
```

File: sidecar/src/connections.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            if prev_count == count
+                && count == 0
+                && prev_time.elapsed() >= min_duration_without_instances
+            {
+                return;
+            }
+
+            prev_count = count;
+            prev_time = tokio::time::Instant::now();
+        }
```
Comment (paullegranddc) [inline sidecar/src/connections.rs:162]: Note for myself to review this better

File: sidecar/src/connections.rs (hunk 1 above)
Comment (pawelchcki) [reply sidecar/src/connections.rs:105]: I need to be able to clone either the counter or notifier separately

File: sidecar/src/connections.rs (hunk 2 above)
Comment (pawelchcki) [reply sidecar/src/connections.rs:162]: :sweat_smile: 
better not ;)

**Pull request closed without merging**